airflow/providers/amazon/aws/operators/batch.py

```diff
@@ -206,8 +206,8 @@ def __init__(
         self.scheduling_priority_override = scheduling_priority_override
         self.array_properties = array_properties
         self.parameters = parameters or {}
-        self.retry_strategy = retry_strategy or {}
-        if not self.retry_strategy.get("attempts", None):
+        self.retry_strategy = retry_strategy
+        if self.retry_strategy is not None and not self.retry_strategy.get("attempts", None):
```

Comment:
```suggestion
        if self.retry_strategy and not self.retry_strategy.get("attempts"):
```

Reply:
Thanks for the suggestion. 

I think in the case where `self.retry_strategy = {}`, the intention of the existing code is to update it to `{"attempts": 1}`. But `bool({})` evaluate to False. So unless I'm mistaken, I don't think your suggestion would work/preserve the existing behavior.

Having that said, that bit of code was introduced because retry_strategy was defaulting to `{}`, causing issue in aws batch. Given now it default to None, we don't have that issue any more (AWS batch can handle None, but can't handle `{}` for retry strategy). So I could just get rid of this. If the user wants to specify a retry_strategy, it's their responsibility to set the `attempts` correctly. We shouldn't be mutating the user input in place anyway.

Reply:
I've removed this for now, I think it makes more sense to not touch the retry_strategy and pass null if not provided. 

Though I'd be curious to hear what @evgenyslab and @ferruzzi think about this.

Reply:
If Batch will accept None then I guess it makes sense.  But if that is the case, then I'm wondering why it was defaulting to {}.  

Shouldn't we consider this a breaking change?  If it used to default to 1 retry and now it does not, that seems like a breaking change to me.

Reply:
It is breaking change but we should treat it as bug fix since the current behavior does not make sense

Reply:
I think just as many people may have been benefiting and enjoying the last defaulting implementation (adding the attempts), I don't see that as so clearly wrong here. The whole point of Airflow operators is to make it easy and convenient for users to build workflows without needing to know the minutiae of the underlying SDK, otherwise everyone would just use taskflow/python operator with the boto sdk directly. We default values on behalf of the user very frequently.

Reply:
@o-nikolas I don't disagree with you, but specifying the default number of attempts explicitly is not necessary.

Prior to https://github.com/apache/airflow/pull/35789 and https://github.com/apache/airflow/pull/35808 the aws `BatchOperator` didn't specify any retryStrategy when submitting the job with boto. That meant it would **use the retry strategy set in the job definition**. The retry strategy specified in the job definition is either specified by the user, or if the user doesn't specify one, it will have one attempt. See "By default, each job is given one attempt to move to either the SUCCEEDED or FAILED job state" [here](https://docs.aws.amazon.com/batch/latest/userguide/job_retries.html)

The new behaviour after these 2 MRs is, if the user doesn't specify the retry policy, to **ignore the job definition retry strategy**. This constitute a breaking change / change in the API behaviour.

Now given when the job definition doesn't have a custom strategy, and the retry strategy is not specified in boto3, it will fallback to the default of 1 attempt, there is no need to specify a explicitly a `{"attempts":1}` in the retry strategy. So it's best to leave it to None. And then we will preserve the previous behaviour.

Reply:
> The retry strategy specified in the job definition is either specified by the user, or if the user doesn't specify one, it will have one attempt. See "By default, each job is given one attempt to move to either the SUCCEEDED or FAILED job state" [here](https://docs.aws.amazon.com/batch/latest/userguide/job_retries.html)

This is a completely different story then, if the new default with no input provided still matches the previous default we had then we have no issue and I see no back-compat issues and the discussion is moot. @vincbeck @ferruzzi agreed?

Reply:
Yep! Agreed! I like the way it is now

Reply:
Ah, that does change things.   Yeah, in that case I agree, this is a bugfix.  Thanks for the clarification, I missed the context of the other two PRs breaking it.

Reply:
Seeing as how Vincent has already approved, and Niko is happy with this conversation, I'll be the second approval and merge it.   Thanks.